plugins/playstation/src/playstation-uri-game-factory.vala

```diff
@@ -0,0 +1,125 @@
+// This file is part of GNOME Games. License: GPLv3
+
+public class Games.PlayStationUriGameFactory : Object, UriGameFactory {
+	public delegate Game GameForUri (string uri) throws Error;
+
+	private const uint HANDLED_URIS_PER_CYCLE = 5;
+
+	private GameForUri game_for_uri;
```

Comment:
We don't need that, directly include the function building the games as a method. Also it's wrong as we don't want to build games from just one URI.